File: two-sum/java/TwoSum.java
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    /**
     * Given an array of integers and a target,
     * return indices of the two numbers that add up to target.
     * Returns empty array if no solution exists.
     */
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (seen.containsKey(complement)) {
                return new int[]{seen.get(complement), i};
            }
            seen.put(nums[i], i);
        }
        return new int[]{};
    }

    public static void main(String[] args) {
        // Quick smoke test
        int[] result = twoSum(new int[]{2, 7, 11, 15}, 9);
        assert result.length == 2 && result[0] == 0 && result[1] == 1 : "Test failed";
        System.out.println("Java: twoSum([2,7,11,15], 9) = [" + result[0] + "," + result[1] + "] — PASS");
    }
}

File: two-sum/go/two_sum.go
package twosum

// TwoSum returns indices of the two numbers that add up to target.
// Returns empty slice if no solution exists.
func TwoSum(nums []int, target int) []int {
	seen := make(map[int]int)
	for i, num := range nums {
		complement := target - num
		if j, ok := seen[complement]; ok {
			return []int{j, i}
		}
		seen[num] = i
	}
	return []int{}
}
